pkg/phlaredb/phlaredb.go

```diff
@@ -459,3 +342,23 @@ func (f *PhlareDB) Flush(ctx context.Context) error {
 	}
 	return oldHead.Flush(ctx)
 }
+
+type blockEviction struct {
+	blockID ulid.ULID
+	err     error
+	evicted bool
+	done    chan struct{}
+}
+
+// Evict removes the given local block from the PhlareDB.
+// Note that the block files are not deleted from the disk.
+// No evictions should be done after and during the Close call.
+func (f *PhlareDB) Evict(blockID ulid.ULID) (bool, error) {
+	e := &blockEviction{
+		blockID: blockID,
+		done:    make(chan struct{}),
+	}
+	f.evictCh <- e
+	<-e.done
+	return e.evicted, e.err
+}
```

Comment:
The idea here is to prevent disk access races: when a block is getting removed from the disk, we need to make sure that no queries will hit it. Given that we also need to manage a global retention policy the method is public